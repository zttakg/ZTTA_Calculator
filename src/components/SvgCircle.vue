<template>
  <circle
    :cx="centerX"
    :cy="centerY"
    :r="radius"
    :fill="fillColor"
    :stroke-width="strokeWidth"
    :stroke="strokeColor"
  />
</template>

<script>
export default {
  name: "SvgCircle",

  props: {
    diameter: {
      type: String,
      default: "300"
    },
    cx: String,
    cy: String,
    strokeWidth: Number,
    strokeColor: String,
    fillColor: String
  },

  computed: {
    radius() {
      return parseInt(this.diameter, 10) / 2;
    },
    centerX() {
      return this.getCenter(this.cx);
    },
    centerY() {
      return this.getCenter(this.cy);
    }
  },

  methods: {
    getCenter(coordinate) {
      const result = coordinate
        ? parseInt(this.radius, 10) + parseInt(coordinate, 10)
        : this.radius;

      if (this.strokeWidth) {
        return result + parseInt(this.strokeWidth, 10);
      }

      return result;
    }
  }
};
</script>
