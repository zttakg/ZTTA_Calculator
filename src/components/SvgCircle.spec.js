import { shallowMount } from "@vue/test-utils";
import SvgCircle from "./SvgCircle";

describe("<SvgCircle /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgCircle);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render circle using `diameter` prop", () => {
    const wrapper = shallowMount(SvgCircle, {
      propsData: {
        diameter: "100"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should move using `cx` and `cy`", () => {
    const wrapper = shallowMount(SvgCircle, {
      propsData: {
        diameter: "100",
        cx: "10",
        cy: "20"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should move using `cx`, `cy` and `strokeWidth` props", () => {
    const wrapper = shallowMount(SvgCircle, {
      propsData: {
        diameter: "100",
        cx: "10",
        cy: "20",
        strokeWidth: 5,
        strokeColor: "#000"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should fill circle using `fillColor` prop", () => {
    const wrapper = shallowMount(SvgCircle, {
      propsData: {
        fillColor: "#000"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
