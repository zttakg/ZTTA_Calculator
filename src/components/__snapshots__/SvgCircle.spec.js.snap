// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgCircle /> component should fill circle using \`fillColor\` prop 1`] = `<circle cx="150" cy="150" r="150" fill="#000"></circle>`;

exports[`<SvgCircle /> component should move using \`cx\` and \`cy\` 1`] = `<circle cx="60" cy="70" r="50"></circle>`;

exports[`<SvgCircle /> component should move using \`cx\`, \`cy\` and \`strokeWidth\` props 1`] = `<circle cx="65" cy="75" r="50" stroke-width="5" stroke="#000"></circle>`;

exports[`<SvgCircle /> component should render circle using \`diameter\` prop 1`] = `<circle cx="50" cy="50" r="50"></circle>`;

exports[`<SvgCircle /> component should render correct without props 1`] = `<circle cx="150" cy="150" r="150"></circle>`;
